docs/AuthConnectAttachTypes.md

```diff
@@ -0,0 +1,291 @@
+- You only need basic connection information (source/destination IP/port)
+- Interface and route information is not relevant to your use case
+
+### Use AUTH_CONNECT Attach Types When:
+- You need **interface information** for authorization decisions (via `bpf_sock_addr_get_interface_type()`)
+- You need **tunnel type** information (via `bpf_sock_addr_get_tunnel_type()`)
+- You need **next-hop interface** details (via `bpf_sock_addr_get_nexthop_interface_luid()`)
+- You need **sub-interface** granularity (via `bpf_sock_addr_get_sub_interface_index()`)
+- You want to **authorize or deny** connections based on complete network context (**no redirection support**)
+- Your policy depends on route-dependent metadata
+
+## Implementation Details
+
+### Program Type
+Both AUTH_CONNECT attach types use the `BPF_PROG_TYPE_CGROUP_SOCK_ADDR` program type, sharing the same context structure and helper functions as other sock_addr programs.
+
+### Verdict Handling
+AUTH_CONNECT programs can return:
+- `BPF_SOCK_ADDR_VERDICT_PROCEED` - Allow the connection to proceed normally
+- `BPF_SOCK_ADDR_VERDICT_PROCEED_HARD` - Allow the connection and skip further authorization checks
+- `BPF_SOCK_ADDR_VERDICT_PROCEED_SOFT` - Allow the connection but continue with additional authorization checks
+- `BPF_SOCK_ADDR_VERDICT_REJECT` - Deny the connection
+
+#### Verdict Types Explained
+
+**PROCEED_HARD vs PROCEED_SOFT:**
+- `PROCEED_HARD` provides an optimization by signaling to the WFP layer that no further authorization checks are needed for this connection, potentially improving performance for trusted connections
+- `PROCEED_SOFT` allows the connection but ensures that other WFP filters and authorization mechanisms continue to evaluate the connection normally
+- `PROCEED` (standard) behaves the same as `PROCEED_SOFT` - allows the connection with normal authorization flow
+
+**Use Cases:**
+- Use `PROCEED_HARD` for connections that have been thoroughly validated and are known to be safe, where bypassing additional checks improves performance
+- Use `PROCEED_SOFT` or `PROCEED` for connections that should be allowed but may still need evaluation by other security mechanisms
+- Use `REJECT` to block connections that violate security policies
+
+**Important:** Redirect functionality is **not supported** at the AUTH_CONNECT layer since route selection has already occurred. Connection redirection can only be performed at the `BPF_CGROUP_INET4_CONNECT` and `BPF_CGROUP_INET6_CONNECT` layers.
+
+### Additional Helper Functions
+AUTH_CONNECT and AUTH_RECV_ACCEPT attach types provide access to additional network layer properties through specialized helper functions:
+
+#### `bpf_sock_addr_get_interface_type(ctx)`
+Returns the network interface type for the connection. Available for AUTH_CONNECT and AUTH_RECV_ACCEPT hooks.
+- **Returns**: Interface type value, or 0 if not available
+- **Use case**: Distinguish between different interface types (e.g., Ethernet, WiFi, VPN)
```

Comment:
"VPN" is not an interface type (in the ifType sense).  "Tunnel" is.

Reply:
Agree. Let me fix that.